### ISSUE TITLE
Project loading is much slower than old version (e.g., 0.2.1) [Critical issue]

### ISSUE THREAD

**Issue by rwangVorbio** [teofiln/gene.editing.dash#7]
@teofiln 
I noticed the current project loading speed in the current version is much slower than the previous version (e.g, 0.2.1). I tested in both R connect and your shiny apps, it currently will need ~ 20 secs to load a project, even if it is the example dataset. I remember previously the time is only ~ 6 sec. Could you check and clarify what is causing this issue? 
**This is indeed a very critical issue, since the only reason we switch from commercial solution to R shiny, is it is faster**.  
![image](https://user-images.githubusercontent.com/100057081/155063420-85346f8d-4076-4d63-a963-1b2b283a82c0.png)

**Comment by rwangVorbio** [comment]
@teofiln , below is the updated loading time of 0.2.6:
![image](https://user-images.githubusercontent.com/100057081/155341580-2af4a920-d4fb-4fe5-96ce-7ac9d91af4a3.png)

All of our team members agree this is a MUST FIX bug, and it should be fixed ASAP.

Please let me know:
1) whether you have any assuring clue on how to fix it
2) whether you can fix this by the end of this week

**Comment by teofiln** [comment]
1. Yes, I have clues.
2. It is possible we can fix it by the end of the week.
Teo

**Comment by teofiln** [comment]
@rwangVorbio 

Please test this instance: https://diatom.shinyapps.io/seq_aln_disabled/ 

I disabled the sequence alignment module. I think the main reason for the slow-down of the application is reshaping or rendering the sequence alignment table. I'll take a closer look at why this might be the case.

**Comment by rwangVorbio** [comment]
@teofiln it is much faster in shinyapps
is this the same version with ver 0.2.6?

**Comment by teofiln** [comment]
Yes, the alignment module is slowing things down. The question is to find what part is causing it and how to speed it up. I am going to run some profiling and be back with more info.

**Comment by teofiln** [comment]
Reshaping the alignment table is a big part of the slowness. Not all of it, but a large part.

I optimized the function to use data table underneath, and we get much better performance:

```
> # dplyr
> tictoc::tic()
> dp <- reshape_alignment_data(.data = .data, .scale_by = "PctReads", .top_n = 3, .n_bp = "30")
> tictoc::toc()
4.573 sec elapsed
> 
> # dtplyr
> tictoc::tic()
> dt <- reshape_alignment_data_dt(.data = .data, .scale_by = "PctReads", .top_n = 3, .n_bp = "30")
> tictoc::toc()
5.205 sec elapsed
> 
> #tidytable
> tictoc::tic()
> tt <- reshape_alignment_data_tt(.data = .data, .scale_by = "PctReads", .top_n = 3, .n_bp = "30")
> tictoc::toc()
0.632 sec elapsed
```

Still needs to be tested, but looks promissing.

**Comment by teofiln** [comment]
@rwangVorbio 

I deployed the updated version: https://diatom.shinyapps.io/gene_editing_dashboard/

Updates:
1. use the faster reshaping function (per above comment)
2. don't reshape the sequence alignment table **unless and until** the user clicks on that tab. Then, print a notification to explain the delay in rendering. 

1 and 2 make loading the project faster and the user can interact with the rest of the tabs immediately and only have to wait _if_ they go to the alignment page.

3. this version also has the fix for the search menu on the project selection page. When you deselect a project (row in the table) the dropdown filters should appear.

None of these have been thoroughly tested. Please let me know if you see any glitches.

**Comment by teofiln** [comment]
@rwangVorbio 

Just made a release for this: https://github.com/teofiln/gene.editing.dash/releases/tag/v0.2.7

Might be a bit early and not tested much. But I think you'll probably want to try it on your RSconnect env.

Let me know how it goes!

**Comment by rwangVorbio** [comment]
@teofiln , I just tried to deploy it, you may want to update the Description file with the information of `tidaytable`:

![image](https://user-images.githubusercontent.com/100057081/155442267-86b522c4-b23e-4f91-9f62-1b35a4dec1b3.png)

**Comment by teofiln** [comment]
It is listed in the description file. But looks like the server does not have it?

**Comment by rwangVorbio** [comment]
@teofiln, good point. I just manually installed `tidaytable`. The new version is online now. Few feedbacks:

1. Thanks for fixing the project search bar issue, nice!

2. it is indeed much faster now, good job!  I noticed the sequence alignment will reload every time it is clicked. It's fine to me and should be fine to other team members considering the improvement of the speed.

3. The order of the alignment in the sequence alignment panel is not consistent with the previous version after loading. In the previous version, the alignment is ordered by the average PctReads across all the samples, so the first row will be the top one alignment shows the highest reads among all the samples. In this version, it seems just randomly ordered?

4. It seems `set alignment` function doesn't work in the real dataset again:
![image](https://user-images.githubusercontent.com/100057081/155445869-844aac7a-a3ef-4798-9e67-7830e109da0f.png)

**Comment by teofiln** [comment]
Hi @rwangVorbio 

re: 3. The `tidytable` version of 'pivot_wider/dcast` does not respect the ordering of the 'names_from` variable. Apparently, it sorts the variable containing the names of the new columns and gives that order of columns in the pivoted table. I switched back to `tidyr::pivot_wider` for this step and the order should be fine now. We have the column with the highest PctReads show up first:

![image](https://user-images.githubusercontent.com/2613403/155641720-062fb548-7d0c-4912-947c-187b05b5f891.png)

Let me know if this is the desired output.

**Comment by teofiln** [comment]
re: 4. It seems set alignment function doesn't work in the real dataset again

I can't reproduce this on my end. But the version I have locally is changed. Let me make another release and lets see if this issue persist after the upgrade.

**Comment by rwangVorbio** [comment]
@teofiln , I just tested the new version 0.2.8, below are feedback:
1. I like the loading spinners, thanks!
2. Caching is good, I would suggest setting the cache into 1Gb
3. the new feature of **flipping X/Y** works perfectly, nice job! (you can close that ticket now)
4. Thanks for the color legend
5. I can see now the alignment is consistent with the previous version
6. **The still does not work in both R connect and your shinyapps.io, below is the screenshot:**
**_R connect:_**
![image](https://user-images.githubusercontent.com/100057081/155648167-6eefc929-6350-460c-a454-e7d96e512fb2.png)

_**shinyapps.io**_
![image](https://user-images.githubusercontent.com/100057081/155648241-c43bbbd0-59d4-4597-b132-f2138026946a.png)

**Comment by teofiln** [comment]
Hi @rwangVorbio 

The current logic for setting alignment length is to check if the reference sequence is < 30 bp in length. If it is, then we enable the trimming from the ends. If not, we render it as is:

https://github.com/teofiln/gene.editing.dash/blob/f981b3bad9e457b432d42c24a2aa124504b8c3f1/R/mod_seq_alignment_fct.R#L4

Should I change the logic to trim _if_ the sequence allows it? If the sequence is 25 bp, to still allow trimming down to 20 and 10? If I remember correctly the original request was with 30 as a threshold.
